tools/lib/h5tools_str.c

```diff
@@ -1180,10 +1180,49 @@ h5tools_str_sprint(h5tools_str_t *str, const h5tool_format_t *info, hid_t contai
                     else if (H5Tequal(type, H5T_STD_REF_DSETREG)) {
                         /* if(nsize == H5R_DSET_REG_REF_BUF_SIZE) */
                         H5TOOLS_DEBUG("H5T_REFERENCE:H5T_STD_REF_DSETREG");
+                        h5tools_str_append(str, H5_TOOLS_DATASET);
+                        h5tools_str_sprint_old_reference(str, container, vp);
                     }
                     else if (H5Tequal(type, H5T_STD_REF_OBJ)) {
                         /* if (nsize == H5R_OBJ_REF_BUF_SIZE) */
+                        /*
+                         * Object references -- show the type and OID of the referenced object.
+                         */
+                        H5O_info1_t oi;
+
                         H5TOOLS_DEBUG("H5T_REFERENCE:H5T_STD_REF_OBJ");
+                        obj = H5Rdereference2(container, H5P_DEFAULT, H5R_OBJECT, vp);
+                        H5Oget_info2(obj, &oi, H5O_INFO_BASIC);
```

Comment:
No error checks?

Reply:
IDK. There was an established code base when I started working on tools, and no design reqs to do error checking beyond that necessary to display info. Maybe because h5dump is a display only program and it would be obvious when it went wrong?